include/subnet_cadmium.hpp

```diff
@@ -0,0 +1,117 @@
+/**
+* Cristina Ruiz Martin
+* ARSLab - Carleton University
+*
+*/
+
+#ifndef _BOOST_SIMULATION_PDEVS_SUBNET_HPP_
+#define _BOOST_SIMULATION_PDEVS_SUBNET_HPP_
+
+#include <cadmium/modeling/ports.hpp>
+#include <cadmium/modeling/message_bag.hpp>
+#include <limits>
+#include <math.h> 
+#include <assert.h>
+#include <memory>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <algorithm>
+#include <limits>
+#include <random>
+
+#include "../include/data_structures/message.hpp"
+
+using namespace cadmium;
+using namespace std;
+
+//Port definition
+struct subnet_defs{
+    struct out : public out_port<message_t> {};
+    struct in : public in_port<message_t> {};
+};
+//This is a meta-model, it should be overloaded for declaring the "id" parameter
+template<typename TIME>
+class Subnet{
+    using defs = subnet_defs; // putting definitions in context
+    public:
+        //Parameters to be overwriten when instantiating the atomic model
+       
+        // default constructor
+        Subnet() noexcept{
+          state.transmiting     = false;
```

Comment:
According to point 3. in coding convention Indentation: 4 spaces

Reply:
Fixed it. Thanks.